astropy/modeling/utils.py

```diff
@@ -4,16 +4,31 @@
 This module provides utility functions for the models package.
 """
 
+from __future__ import annotations
+
 import warnings
 
 # pylint: disable=invalid-name
 from collections import UserDict
 from inspect import signature
+from typing import TYPE_CHECKING, overload
 
 import numpy as np
 
 from astropy import units as u
 
+if TYPE_CHECKING:
+    from collections.abc import Sequence
+    from typing import SupportsFloat, TypeVar
+
+    from numpy.typing import NDArray
+
+    from astropy.units import Quantity
```

Comment:
`Quantity` had to be imported in an `if TYPE_CHECKING:` block to avoid import loops within `units`, but if this code is moved to modeling then it should be possible to use `u.Quantity` in annotations.
```suggestion
```

Reply:
Yes, makes sense. Done.